extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -79,14 +85,25 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void initialize() {
         GatewayConfig configuration = getConfigAs(GatewayConfig.class);
-        if (configuration.host == null || configuration.host.isEmpty()) {
+
+        if (isNullOrEmpty(configuration.host)) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                     "Host must be specified in the configuration!");
             return;
         }
-        if (configuration.code == null || configuration.code.isEmpty()) {
+        if (isNullOrEmpty(configuration.code)) {
+            if (isNullOrEmpty(configuration.username) || isNullOrEmpty(configuration.userkey)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Either security code or username and pre-shared key must be provided in the configuration!");
+                return;
+            }
+        } else {
+            configuration = getUsernameAndKey(configuration);
```

Comment:
imho, you must not do an assignment here

Reply:
The other option is calling ```configuration = getConfigAs(GatewayConfig.class);``` there (and not in the ```getUsernameAndKey``` ) but anyway - the new configuration has to be retrieved again